### PR TITLE
Travel perk category mapping

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new section for "Category Mapping" in the advanced settings, allowing dynamic configuration.
  - Enhanced the payment profile settings to use more intuitive field names, improving user experience.
- **Enhancements**
  - Updated the Travelperk integration to support dynamic content and branding in the "Customize the Purpose field" section.
  - Improved form handling and mapping logic in both advanced settings and payment profile settings, ensuring a smoother user interaction.
- **Bug Fixes**
  - Corrected spelling mistakes in method names and endpoints related to "advanced_settings" in the Travelperk service, ensuring consistency and reliability.
- **Style**
  - Adjusted styling for the footer section in the advanced settings component for better visual appeal.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts [hunk 1]
```diff
@@ -1,6 +1,15 @@
-import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { FormArray, FormControl, FormGroup, Validators } from "@angular/forms";
 import { TravelPerkExpenseGroup } from "../../enum/enum.model";
-import { SelectFormOption } from "../../common/select-form-option.model";
+import { SelectFormLabel, SelectFormOption } from "../../common/select-form-option.model";
+import { TravelperkDestinationAttribuite } from "../travelperk.model";
+import { DefaultDestinationAttribute } from "../../db/destination-attribute.model";
+
+interface StringDictionary {
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts:7]: better name, TravelperkCategoryMapping

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts [hunk 2]
```diff
@@ -34,9 +76,48 @@ export class TravelperkAdvancedSettingModel {
...
-    static mapAPIResponseToFormGroup(advancedSettings: TravelperkAdvancedSettingGet | null): FormGroup {
+    static createFormGroup(data: TravelperkAdvancedSettingArray): FormGroup {
+        return new FormGroup ({
+            destinationName: new FormControl(data.destination_name || ''),
+            sourceName: new FormControl(data.source_name || null)
+        });
+    }
+
+    static constructFormArray(arrayData: TravelperkAdvancedSettingGet, source_fields: TravelperkDestinationAttribuite[]): FormGroup[] {
+        const resultentArray:FormGroup[] = [];
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts:87]: mappingPayload

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts [hunk 3]
```diff
@@ -34,9 +76,48 @@ export class TravelperkAdvancedSettingModel {
...
+        const destination = this.getDefaultCategory();
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts:88]: defaultCategories

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts [hunk 4]
```diff
@@ -34,9 +76,48 @@ export class TravelperkAdvancedSettingModel {
...
+            resultentArray.push(this.createFormGroup(data));
+        });
+        return resultentArray;
+    }
+
+    static constructCategoryMapping(categoryMappingFieldArray: TravelperkAdvancedSettingFormArray[]) {
+        const data:StringDictionary = {};
+        categoryMappingFieldArray.forEach((item: TravelperkAdvancedSettingFormArray) => {
+            data[item.destinationName.value] = {
+                "name": item.sourceName?.value ? item.sourceName.value : '',
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts:108]: single quote for consistency

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts [hunk 5]
```diff
@@ -34,9 +76,48 @@ export class TravelperkAdvancedSettingModel {
...
+                "id": item.sourceName?.source_id ? item.sourceName.source_id : ''
+            };
+        });
+
+        return data;
+    }
+
+    static mapAPIResponseToFormGroup(advancedSettings: TravelperkAdvancedSettingGet | null, sourceOptions: TravelperkDestinationAttribuite[]): FormGroup {
         const defaultMemoOptions: string[] =['trip_id', 'trip_name', 'traveler_name', 'booker_name', 'merchant_name'];
+        const categoryMappingsArray = advancedSettings ? this.constructFormArray(advancedSettings, sourceOptions) : [] ;
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-advanced-settings.model.ts:118]: categoryMappings

File: src/app/core/services/travelperk/travelperk.service.ts [hunk 6]
```diff
@@ -81,6 +81,10 @@ export class TravelperkService {
     return this.apiService.get(`/orgs/${this.orgId}/travelperk/sync_payment_profile/`,  {});
   }
 
+  syncCategories(): Observable<{}> {
```
Comment (ashwin1111) [inline src/app/core/services/travelperk/travelperk.service.ts:84]: @cacheable()

File: src/app/shared/components/configuration/configuration-mapping-fields/configuration-mapping-fields.component.ts [hunk 7]
```diff
@@ -21,9 +22,9 @@ export class ConfigurationMappingFieldsComponent implements OnInit {
 
   @Input() destinationPlaceholderFieldText: string;
 
-  @Input() sourceFieldOptions: string[] | SelectFormOption[];
+  @Input() sourceFieldOptions: string[] | SelectFormOption[] | TravelperkDestinationAttribuite[];
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-mapping-fields/configuration-mapping-fields.component.ts:25]: any[];

File: src/app/shared/components/configuration/configuration-mapping-fields/configuration-mapping-fields.component.ts [hunk 8]
```diff
@@ -21,9 +22,9 @@ export class ConfigurationMappingFieldsComponent implements OnInit {
...
 
-  @Input() destinationFieldOptions: string[] | TravelperkPaymentProfileSettingGet[];
+  @Input() destinationFieldOptions: string[] | TravelperkPaymentProfileSettingGet[] | TravelperkDestinationAttribuite[] | SelectFormOption[];
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-mapping-fields/configuration-mapping-fields.component.ts:27]: any[]

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**